service/resource/deployment/types.go

```diff
@@ -0,0 +1,45 @@
+package deployment
+
+import providerv1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+
+func newNodes(azureNodes []providerv1alpha1.AzureConfigSpecAzureNode) []node {
+	var ns []node
+	for _, n := range azureNodes {
+		ns = append(ns, newNode(n))
+	}
+	return ns
+}
+
+type node struct {
+	providerv1alpha1.AzureConfigSpecAzureNode
+
+	// OSImage is the vm OS image object.
+	OSImage nodeOSImage `json:"osImage" yaml:"osImage"`
+}
+
+func newNode(azureNode providerv1alpha1.AzureConfigSpecAzureNode) node {
+	return node{
+		AzureConfigSpecAzureNode: azureNode,
+		OSImage:                  newNodeOSImage(),
+	}
+}
+
+type nodeOSImage struct {
+	// Offer is the image offered by the publisher (e.g. CoreOS).
+	Offer string `json:"offer" yaml:"offer"`
+	// Publisher is the image publisher (e.g GiantSwarm).
+	Publisher string `json:"publisher" yaml:"publisher"`
+	// SKU is the image SKU (e.g. Alpha).
+	SKU string `json:"sku" yaml:"sku"`
+	// Version is the image version (e.g. 1465.7.0).
+	Version string `json:"version" yaml:"version"`
+}
+
+func newNodeOSImage() nodeOSImage {
```

Comment:
I would like to have either a good godoc on this because of the hard coded information within the function or constants for the magic strings which are documented properly.

Reply:
I'm afraid it will be difficult to provide better documentation. Even azure official doc is pretty generic: https://docs.microsoft.com/en-us/azure/templates/microsoft.compute/virtualmachines#ImageReference

I will include this link in the godoc though.

Reply:
The important thing for me here is to have the used CoreOS version in the godoc.